loopchain/blockchain/blocks/v0_3/block_verifier.py

```diff
@@ -140,6 +157,9 @@ def verify_leader_votes(self, block: 'Block', prev_block: 'Block',  reps: Sequen
         else:
             prev_block_header: BlockHeader = prev_block.header
             if prev_block_header.next_leader != block.header.peer_id:
+                if prev_block_header.next_leader not in reps and block.header.peer_id == reps[0]:
+                    # prep term changed!
+                    return
                 exception = RuntimeError(f"Block({block.header.height}, {block.header.hash.hex()}, "
                                          f"Leader({block.header.peer_id.hex_xx()}), "
                                          f"Expected({prev_block_header.next_leader.hex_xx()}).\n "
```

Comment:
Calling too much times `prev_block_header.next_leader`
(Up to your choice. I'm not sure this increases readability)
```python
            expected_current_leader = prev_block.header.next_leader
            if expected_current_leader != block.header.peer_id:
                if expected_current_leader not in reps and block.header.peer_id == reps[0]:
                    # prep term changed!
                    return
                exception = RuntimeError(f"Block({block.header.height}, {block.header.hash.hex()}, "
                                         f"Leader({block.header.peer_id.hex_xx()}), "
                                         f"Expected({expected_current_leader.hex_xx()}).\n "
```

Reply:
I think not bad here.